subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1956,11 +1956,11 @@ static int lwm2m_read_handler(struct lwm2m_engine_obj_inst *obj_inst,
 		case LWM2M_RES_TYPE_OPAQUE:
 			break;
 
-		/* TODO: handle multi count for string? */
 		case LWM2M_RES_TYPE_STRING:
 			engine_put_string(&msg->out, &msg->path,
-					  (u8_t *)data_ptr,
-					  strlen((u8_t *)data_ptr));
+					  ((u8_t *)data_ptr + i*data_len),
+					  strlen(((u8_t *)data_ptr +
+							  i*data_len)));
```

Comment:
spaces around `*` in two lines above

Reply:
This logic was removed in favor of each resource instance having it's own data_ptr (as it should be).